domain/src/main/java/com/thoughtworks/go/config/materials/AbstractMaterial.java

```diff
@@ -235,4 +238,8 @@ public boolean ignoreForScheduling() {
         //overridden in DependencyMaterial. Other materials don't support this flag
         return false;
     }
+
+    public ValidationBean checkConnection(final SubprocessExecutionContext execCtx) {
+        throw new UnprocessableEntityException(String.format("'checkConnection' cannot be performed on material of type %s", type));
```

Comment:
At this level, I would prefer `UnsupportedOperationException` instead of `UnprocessableEntityException`